src/transformers/models/gemma2/modeling_gemma2.py

```diff
@@ -567,7 +568,8 @@ def forward(
         if self.is_sliding and attention_mask is not None:  # efficient SDPA and no padding
             # Flash-attn is a 2D tensor
             if self.config._attn_implementation == "flash_attention_2":
-                attention_mask = attention_mask[:, -self.sliding_window :]
+                if past_key_value is not None:  # when decoding
+                    attention_mask = attention_mask[:, -self.sliding_window :]
```

Comment:
Ah that looks better yeah, when you have the first forward pass you nee the attention mask, full but have to use `self.sliding_window` when calling attention, but then when you decode the mask is too big, and the kv cache is sliding_window -1